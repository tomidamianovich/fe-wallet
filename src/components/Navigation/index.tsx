import React from 'react';
import './styles/index.scss';
import NavItem from '../../components/NavItem';

type Link = { 
  name: string, 
  to: string,
  icon: any,
};

type Props = {
  links: Link[];
};

const Navigation: React.FC<Props> = ({ links }) => 
  <nav className="fe-wallet__content__navigation">
    { 
      links && 
      <ul className="fe-wallet__content__navigation__list">
        { links?.map( (link, index) => <NavItem {...link} key={index} />) }
      </ul> 
    }
    <div className="fe-wallet__content__navigation__logout">
      <NavItem name="Salir" icon="logout" to="logout" />
    </div>
  </nav>
;

export default Navigation;