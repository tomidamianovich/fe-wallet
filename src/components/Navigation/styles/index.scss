@import "../../../containers/App/styles/colours.scss";
@import "../../../containers/App/styles/responsiveness.scss";

.fe-wallet__content__navigation {
  background-color: $secondary-base-color;
  min-width: 256px;
  margin: 0;
  padding: 0;
  height: 100%;

  &__list {
    margin: 0;
    padding: 0;
    list-style: none;
    text-decoration: none;
  }

  &__logout {
    position: absolute;
    bottom: 0;
    padding: 25px 0;
  }

  @media screen and (max-width: $sm) {
    display: none;
  }
}