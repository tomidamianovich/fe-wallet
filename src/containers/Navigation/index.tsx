import React from 'react';
import './styles/index.scss';
import { NavLink } from "react-router-dom";

type Props = {
  links: string[];
};

type LinkItemProps = {
  to: string;
  name: string;
};

const LinkItem = ({to, name}: LinkItemProps) =>  
  <li>
    <NavLink 
      to={to}
      className={({ isActive }) => `fe-wallet__content__navigation__list__item ${isActive && 'fe-wallet__content__navigation__list__item--active'}`}
    >
      {name}
    </NavLink>
  </li>;

const Navigation: React.FC<Props> = ({ links }) => 
  <nav className="fe-wallet__content__navigation">
    <ul className="fe-wallet__content__navigation__list">
      { links.map( (link, index) => <LinkItem to={link} name={link} key={index} />) }
    </ul>
  </nav>
;

export default Navigation;