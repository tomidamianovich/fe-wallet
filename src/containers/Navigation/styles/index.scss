@import "../../../containers/App/styles/colours.scss";
@import "../../../containers/App/styles/responsiveness.scss";

.fe-wallet__content__navigation {
  background-color: $secondary-base-color;
  width: 256px;
  margin: 0;
  padding: 0;
  height: 100%;

  &__list {
    margin: 0;
    padding: 0;
    list-style: none;
    text-decoration: none;
    
    &__item {
      color: black;
      text-decoration: none;
      &--active {
        font-weight: 700;
      }
    }
  }

  @media screen and (max-width: $sm) {
    display: none;
  }
}