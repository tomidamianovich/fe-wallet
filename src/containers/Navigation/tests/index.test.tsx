import { render } from '@testing-library/react';
import Navigation from '../index';
import { BrowserRouter } from "react-router-dom";

test('renders Navigation main without items', () => {
  const { asFragment } = render(<BrowserRouter><Navigation links={['']}/></BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});

test('renders Navigation main with items', () => {
  const { asFragment } = render(<BrowserRouter><Navigation links={['link 1', 'link 2', 'link 3']}/></BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});