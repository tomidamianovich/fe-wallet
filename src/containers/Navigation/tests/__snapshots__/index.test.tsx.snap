// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navigation main with items 1`] = `
<DocumentFragment>
  <nav
    class="fe-wallet__content__navigation"
  >
    <ul
      class="fe-wallet__content__navigation__list"
    >
      <li>
        <a
          class="fe-wallet__content__navigation__list__item false"
          href="/link 1"
        >
          link 1
        </a>
      </li>
      <li>
        <a
          class="fe-wallet__content__navigation__list__item false"
          href="/link 2"
        >
          link 2
        </a>
      </li>
      <li>
        <a
          class="fe-wallet__content__navigation__list__item false"
          href="/link 3"
        >
          link 3
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`renders Navigation main without items 1`] = `
<DocumentFragment>
  <nav
    class="fe-wallet__content__navigation"
  >
    <ul
      class="fe-wallet__content__navigation__list"
    >
      <li>
        <a
          aria-current="page"
          class="fe-wallet__content__navigation__list__item fe-wallet__content__navigation__list__item--active"
          href="/"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
